docs/dev/reader-concurrency-semaphore.md

```diff
@@ -59,6 +59,7 @@ Permits are in one of the following states:
 * `active/await` - a previously `active/need_cpu` permit, which needs something other than CPU to proceed, it is waiting on I/O or a remote shards, other permits can be admitted while the permit is in this state, pending resource availability;
 * `inactive` - the permit was marked inactive, it can be evicted to make room for admitting more permits if needed;
```

Comment:
> Being in this state means a permit was removed from the wait list
> and its internal timer was canceled.

Why do we need this state, rather than just remove the query containing the permit?